presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunction.java

```diff
@@ -138,7 +139,7 @@ private boolean matchesParameterAndReturnTypes(Method method, List<Type> resolve
             if (!methodParameterJavaTypes[methodParameterIndex].equals(type)) {
                 return false;
             }
-            methodParameterIndex += nullConvention == USE_NULL_FLAG ? 2 : 1;
+            methodParameterIndex += (nullConvention == USE_NULL_FLAG || nullConvention == BLOCK_POSITION) ? 2 : 1;
```

Comment:
What about the following:

Make `NullConvention` enum to be 

```java
    public enum NullConvention
    {
        RETURN_NULL_ON_NULL(1),
        USE_BOXED_TYPE(1),
        USE_NULL_FLAG(2),
        BLOCK_POSITION(2),
        /**/;

        private final int parameterCount;

        NullConvention(int methodParameterCount)
        {
            this.parameterCount = methodParameterCount;
        }

        public int getParameterCount()
        {
            return parameterCount;
        }
    }
```

Thus here you can do 
```java
    methodParameterIndex += nullConvention.getParameterCount();
```